chore(docs): add authz docs (#1216)
emqtt/emq-docs-cn: en_US/security/authz/authz.md

| mysql              | [Authorization with MySQL database as ACL storage](./mysql.md)                  |
| postgresql         | [Authorization with PostgreSQL database as ACL storage](./postgresql.md)        |
| mongodb            | [Authorization with MongoDB database as ACL storage](./mongodb.md)            |
| redis              | [Authorization with Redis database as ACL storage](./redis.md)                |
| http               | [Authorization using external HTTP API to determine permissions](./http.md)   |
| file               | [Authorization using static ACL configured in a file](./file.md)              |

Each authorizer has its own configuration options.

Example:

```
{
    enable => true

    type = mysql
    database = "mqtt"
    username = "root"
    password = "public"

    query = "SELECT permission, action, topic FROM acl WHERE username = ${username}"
    server = "10.12.43.12:3306"
}
```

## Authorization Chain

Configured authorizers form a global chain. When a client makes a publish/subscribe request, authorizers are
sequentially used to find access lists for the client. If an authorizer finds ACL rules, the request is checked

> against them and gets accepted/rejected. If ACL rules are not found, then the next authorizer from the chain is used.

Reviewer: ```suggestion
against them and gets allowed/denied. If ACL rules are not found, then the next authorizer from the chain is used.
```

If no authorizer finds any ACL rules, then the default permission is applied.

The first element of the chain is always the implicit authorizer that matches the request against the ACL rules

> returned by the authentication (if there are any).

Reviewer: not sure I can understand this.
could you help to explain it?

Author: I tried to simplify that and added a separate section.

Author: https://github.com/emqx/emqx-docs/pull/1216/commits/da09414d538babb98c5485be64969292ce731efe

## ACL Cache

If a client sends requests intensively, it may be resource-consuming to fetch ACL rules for each request and match it
against the rules. Therefore, ACL cache may be enabled to cache authorization results for a particular time.

::: warning

> Caching improves performance significantly, so adjusting the default values to the relevant ones is essential.

Reviewer: ```suggestion
Caching improves performance significantly, so adjusting the default values to the relevant ones is essential.
You can find more information about cache configuration below.
```

:::

## Authorization placeholders

Many authorizers allow using _placeholders_ in their configuration.
Placeholders work as template variables filled with the corresponding

> client's information when performing authentication.

Reviewer: ```suggestion
Placeholders work as template variables filled with the corresponding runtime information.

### Placeholders in configurations

Configuration templates can be filled with client information such ad client ID and user name at runtime.
```

The following placeholders are available:
* `${clientid}` — clientid of the client.
* `${username}` — username of the client.
* `${peerhost}` — client IP address.
* `${cert_subject}` — subject of client's TLS certificate, valid only for TLS connections.
* `${cert_common_name}` common name of client's TLS certificate, valid only for TLS connections.

Example of use in Redis authorizer:

```
{
  enable = true
  type = redis

  ... other parameters ...

  cmd = "HGETALL mqtt_user:${username}"
}
```

> ## Topic placeholders

Reviewer: ```suggestion
### Placeholders in ACL rules
```
